src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2531,6 +2562,10 @@ private Process StartWithCreateProcess(ProcessStartInfo startinfo)
                 {
                     pinnedEnvironmentBlock.Free();
                 }
+                else
+                {
+                    ProcessNativeMethods.DestroyEnvironmentBlock(AddressOfEnvironmentBlock);
```

Comment:
should this be in a finally?

Reply:
It is already in finally. Or I don't understand you question.